Source/AtomicJS/Javascript/JSDebugger.cpp

```diff
@@ -0,0 +1,862 @@
+#include "JSVM.h"
+#include "JSDebugger.h"
+
+// forward declaratons
+
+static void duk_trans_socket_init(void);
+static void duk_trans_socket_waitconn(void);
+static duk_size_t duk_trans_socket_read_cb(void *udata, char *buffer, duk_size_t length);
+static duk_size_t duk_trans_socket_write_cb(void *udata, const char *buffer, duk_size_t length);
+static duk_size_t duk_trans_socket_peek_cb(void *udata);
+static void duk_trans_socket_read_flush_cb(void *udata);
+static void duk_trans_socket_write_flush_cb(void *udata);
+static void debugger_detached (duk_context* ctx, void *udata);
+static void duk_trans_socket_finish(void);
+
+namespace Atomic
+{
+
+JSDebugger* JSDebugger::instance_ = NULL;
+
+JSDebugger::JSDebugger(Context* context) : Object(context)
+{
+    instance_ = this;
+    this->AutoReconnect = false;
+}
+
+JSDebugger::~JSDebugger()
+{
+    instance_ = NULL;
+}
+
+void JSDebugger::Shutdown() const
+{
+    duk_trans_socket_finish();
```

Comment:
Should ```Shutdown``` ensure that auto reconnect is turned off, that way don't need to depend on calling sites setting it?